pkg/tagger/collectors/kubernetes_main.go

```diff
@@ -27,27 +28,36 @@ type KubeServiceCollector struct {
 	kubeUtil  *kubelet.KubeUtil
 	apiClient *apiserver.APIClient
 	infoOut   chan<- []*TagInfo
-
+	dcaClient *clusteragent.DCAClient
 	// used to set a custom delay
 	lastUpdate time.Time
 	updateFreq time.Duration
 }
 
-// Detect tries to connect to the kubelet
-// TODO refactor when we have the DCA
+// Detect tries to connect to the kubelet and the API Server if the DCA is not used or the DCA.
 func (c *KubeServiceCollector) Detect(out chan<- []*TagInfo) (CollectionMode, error) {
 	if config.Datadog.GetBool("kubernetes_collect_service_tags") == false {
 		return NoCollection, fmt.Errorf("collection disabled by the configuration")
 	}
 
-	var err error
+	var err, errDCA error
 	c.kubeUtil, err = kubelet.GetKubeUtil()
 	if err != nil {
 		return NoCollection, err
 	}
-	c.apiClient, err = apiserver.GetAPIClient()
-	if err != nil {
-		return NoCollection, err
+	// if no DCA or can't communicate with the DCA run the local service mapper.
+	if config.Datadog.GetBool("cluster_agent") {
+		c.dcaClient, errDCA = clusteragent.GetClusterAgentClient()
+		if errDCA != nil {
+			log.Errorf("Could not initialise the communication with the DCA, falling back to local service mapping: %s", errDCA.Error())
```

Comment:
lazy tip: Usually I used `%v` with just `err` instead of `%s` + `err.Error()`
:nail_care:

Reply:
Interesting, do you know if this affects the perf ? I have been using %s for a while, but if there is a consensus I'll change